src/types.ts

```diff
@@ -427,4 +450,34 @@ export class mIOToken extends PositiveFiniteInteger {
   constructor(value: number) {
     super(value);
   }
+
+  multiply(multiplier: mIOToken | number): mIOToken {
```

Comment:
allowing # here to easily multiply tokens by decimals and force them to be rounded down. particularly useful when dealing with weights and distributions.